cnxarchive/database.py

```diff
@@ -111,7 +112,7 @@ def initdb(settings):
 
 def get_module_ident_from_ident_hash(ident_hash, cursor):
     """Returns the moduleid for a given ``ident_hash``."""
-    uuid, (mj_ver, mn_ver) = split_ident_hash(ident_hash, split_version=True)
+    uuid, (mj_ver, mn_ver), id_type = split_ident_hash(ident_hash, split_version=True)
```

Comment:
Why do we need the id_type returned from every occurrence of the split?

I have a feeling this is going to **rot into a pungent code smell**. Perhaps we should treat all ident-hashs as invalid if they aren't the short version?

We will only handle the uuid based ids in known places, right? One of those being during route matching for redirecting uuid based URIs to short-id based URIs.